### ISSUE TITLE
"Ask Before Moving Undated Items" always overwrites the date when checked, should probably be a setting.

### ISSUE THREAD

**Issue by WaveSmashies** [deckerst/aves#1357]
**Describe the bug**
If you enable "Don't ask again" regarding saving the date on files you move, it will not save or respect your "Save Dates" setting. So if you tell it not to ask again, your files will have their date and time overwritten by moving them from then on. This is a little annoying and should probably get a setting as well.

**To Reproduce**
1. Settings
2. Navigation
3. Confirmation Dialogs
4. Ask Before Moving Undated Items
5. You can also move a file and it will ask if you want to save the dates. If you check Do Not Ask Again, it will NOT save the dates with every move from then on.

**Expected behavior**
I kinda expected it to save the dates if I selected save dates before chosing don't ask again. It does not lol.

**Additional context**
Sorry if this is something which is already fixed or a setting that I just missed. I poked around all I could but didn't see it. Appreciate it!

**Comment by deckerst** [comment]
Either there is a confusion about what we mean by "save dates", or there is indeed a bug. So let me clarify:
- the `file date` cannot be controlled by Aves and it will be changed when doing file operations,
- the `metadata date` can be edited by Aves, and it is the date preferably used by the app (over the file date),
- the dialog you mention appears when the metadata date is missing, so it asks the user whether to save the file date in the metadata before doing the file operation (which will modify the file date in any case).

Do you confirm there is a bug, given the explanation above?

**Comment by WaveSmashies** [comment]
When Save Dates is disabled, then after the file is moved you'll find it changed the date and time to when you moved it. If Save Dates is enabled, it keeps the date and time which you created or saved the file. What I'm saying is I'd prefer Save Dates be permanently enabled. But if you enable Save Dates before chosing Don't Ask Again, it will not Save Dates when you move files later. That might be a missing feature but I put it in as a bug because I was expecting it to Save Dates because I had enabled that before chosing not to ask again.

**Comment by deckerst** [comment]
> But if you enable Save Dates before chosing Don't Ask Again, it will not Save Dates when you move files later.

I agree it's a bug, it should do as you expect. I'll check on my device.

Also, it's possible it doesn't work for you because you move files for which metadata editing is not supported.

**Comment by WaveSmashies** [comment]
I'm talking about typical images. PNG, JPG and JPEG, sometimes MP4 and MOV but by far most often PNG and JPG.

**Comment by deckerst** [comment]
I've just tried with a PNG, and it saved the date as expected. 

Could you provide a sample file for which it fails for you? (Preferably a JPG or PNG)

**Comment by deckerst** [comment]
Please provide the system info and logs. You can get them from the app About > Bug Report. Best is to export the logs just after you get the issue.

**Comment by deckerst** [comment]
No worries. If it happens again with a basic JPG or PNG, please provide sample and logs. Otherwise, for MP4 and the likes, it's probably because metadata editing fails (and that's a known limitation by format).

**Comment by WaveSmashies** [comment]
Does selecting Don't Ask Again also save the state of Save Dates when you enable it? If it does then this shouldn't really be a problem.

Also, I checked and I see I can't use Remove Metadata on PNG files, but I CAN manually change the date and time, or even remove the date and time via Edit > Edit Date And Time. Is that a potential workaround?
Whatever's going on, I'd like it to stop happening lol. Moving a collection of images which are intentionally ordered by date, only for them to just shuffle amongst themselves for no reason is kinda annoying.

**Comment by deckerst** [comment]
> Does selecting Don't Ask Again also save the state of Save Dates when you enable it? If it does then this shouldn't really be a problem.

Yes.

> Moving a collection of images which are intentionally ordered by date, only for them to just shuffle amongst themselves for no reason is kinda annoying.

Indeed, but the only way to move forward is with logs and samples.
